From cc61e54f1514fdab357fa84034f4c51fc1bad191 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marek=20Marczykowski-G=C3=B3recki?=
 <marmarek@invisiblethingslab.com>
Date: Wed, 25 Aug 2021 02:41:51 +0200
Subject: [PATCH] gnttab: disable grant tables v2 by default
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
Organization: Invisible Things Lab
Cc: Marek Marczykowski-Górecki <marmarek@invisiblethingslab.com>

They are not used anywhere, and are significant attack surface.

Signed-off-by: Marek Marczykowski-Górecki <marmarek@invisiblethingslab.com>

-----------
diff --git a/xen/common/grant_table.c b/xen/common/grant_table.c
index 4c742cd8..de14e6ee 100644
--- a/xen/common/grant_table.c
+++ b/xen/common/grant_table.c
@@ -176,7 +176,7 @@ static int parse_gnttab_max_maptrack_frames(const char *arg)
 }
 
 #ifndef GNTTAB_MAX_VERSION
-#define GNTTAB_MAX_VERSION 2
+#define GNTTAB_MAX_VERSION 1
 #endif
 
 unsigned int __read_mostly opt_gnttab_max_version = GNTTAB_MAX_VERSION;
-----------------
